### PR TITLE
Make `_all` field the default search field.

### PR DESCRIPTION
**Pull request by mansu**
Make `_all` field the default search field so we don't have to enter it every time we have to do a full text search on logs.

### CODE REVIEW

**Comment by vthacker** [comment]
I don't think this is quite right. What if all field is disabled? This PR doesn't take into account of that

Secondly, what if all field was enabled for some chunks and then disabled for some chunks?

**Comment by vthacker** [comment]
> I don't think this is quite right. What if all field is disabled? This PR doesn't take into account of that
> 
> Secondly, what if all field was enabled for some chunks and then disabled for some chunks?

The same questions would apply to the default Message field that we use today. So for now let's merge

File: kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java [hunk 1]
```diff
@@ -382,26 +382,112 @@ public void testFullTextSearch() {
         makeMessageWithIndexAndTimestamp(1, "apple", TEST_DATASET_NAME, time.plusSeconds(4));
     msg1.addProperty("field1", "1234");
     strictLogStore.logStore.addMessage(msg1);
+    strictLogStore.logStore.commit();
+    strictLogStore.logStore.refresh();
+    // Search using _all field.
+    assertThat(
```
Comment (vthacker) [inline kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:388]: can we add a test with multiple words with spaces and some numbers etc ?

Comment (bryanlb) [reply kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:388]: Let's also go ahead and add tests for empty string and wildcard as well.

Comment (mansu) [reply kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:388]: Done.

File: kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java [hunk 2]
```diff
@@ -382,26 +382,157 @@ public void testFullTextSearch() {
...
+                .search(TEST_DATASET_NAME, "", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(3);
 
-    SearchResult<LogMessage> babyInAll =
-        strictLogStore.logSearcher.search(TEST_DATASET_NAME, "_all:baby", 0, MAX_TIME, 1000, 1);
-    assertThat(babyInAll.hits.size()).isEqualTo(2);
+    // Currently, returns empty since we don't parse wild card queries.
+    // TODO: One we start parsing wild card queries this should return 3.
```
Comment (bryanlb) [inline kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:502]: This isn't accurate - for fields that are analyzed wildcard queries do work today. You can see this if you query for "app*"  in this test you should have results for "apple". Additionally if this is just "*" it returns the entire dataset when the field is analyzed.

**Review by bryanlb — CHANGES_REQUESTED** [review]

File: kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java (hunk 2 above)
Comment (mansu) [reply kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java]: Good catch. Updated the tests for wildcard queries. * throws an exception. If we want * to not throw an exception, we should update the query parser to convert that to empty query.

File: kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java [hunk 3]
```diff
@@ -424,15 +580,38 @@ public void testDisabledFullTextSearch() {
...
+                .size())
+        .isEqualTo(0);
+
+    assertThat(
+            strictLogStoreWithoutFts
+                .logSearcher
+                .search(TEST_DATASET_NAME, "1234", 0, MAX_TIME, 1000, 1)
+                .hits
+                .size())
+        .isEqualTo(0);
```
Comment (bryanlb) [inline kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:614]: Probably would be good to include the test asserts in the full text search disabled case as well, and not just the enabled one.

Comment (mansu) [reply kaldb/src/test/java/com/slack/kaldb/logstore/search/LogIndexSearcherImplTest.java:639]: Most of the additional asserts are related to how query parser handles the queries and not the all field. So, those tests are not as relevant here. However, added them here for completeness. 

We also have simpler higher level test at the chunk level, so we are covered there.

**Review by bryanlb — APPROVED** [review]
